scripts/fetch_data.py

```diff
@@ -0,0 +1,274 @@
+       counterparts
+    """
+    args = {}
+
+    ce_variable = {
+        'temp': {'min': 'tasmin', 'max': 'tasmax'},
+        'prec': 'pr',
+        'dg05': 'gdd',
+        'nffd': 'fdETCCDI',
+        'pass': 'prsn',
+        'dl18': 'hdd'
+    }[variable]
+    args['variable'] = ce_variable
+
+    ce_time, ce_timescale = {
+        'ann': ['0', 'yearly'],
+        'djf': ['0', 'seasonal'],
+        'mam': ['1', 'seasonal'],
+        'jja': ['2', 'seasonal'],
+        'son': ['3', 'seasonal'],
+        'jan': ['0', 'monthly'],
+        'feb': ['1', 'monthly'],
+        'mar': ['2', 'monthly'],
+        'apr': ['3', 'monthly'],
+        'may': ['4', 'monthly'],
+        'jun': ['5', 'monthly'],
+        'jul': ['6', 'monthly'],
+        'aug': ['7', 'monthly'],
+        'sep': ['8', 'monthly'],
+        'oct': ['9', 'monthly'],
+        'nov': ['10', 'monthly'],
+        'dec': ['11', 'monthly'],
+    }[time_of_year]
+    args['time'] = ce_time
+    args['timescale'] = ce_timescale
+
+    ce_cell_method = {
+        'iastddev': 'standard_deviation',
+        'iamean': 'mean'
+    }[temporal]
+    args['cell_method'] = ce_cell_method
+
+    ce_spatial = {
+        's0p': 'min',
+        's100p': 'max',
+        'smean': 'mean'
+    }[spatial]
+    args['spatial'] = ce_spatial
+
+    ce_percentile = {
+        'e25p': 25,
+        'e75p': 75,
+        'hist': 100
+    }[percentile]
+    args['percentile'] = ce_percentile
+
+    if percentile == 'hist':
+        emission = percentile
+    else:
+        emission = variable
+    ce_emission = {
+        'temp': 'historical,rcp85',
+        'prec': 'historical,rcp85',
+        'dg05': 'historical,rcp85',
+        'nffd': 'historical, rcp85',
+        'pass': 'historical,rcp85',
+        'dl18': 'historical,rcp85',
+        'hist': ''  # historical has no emission scenario
+    }[emission]
+    args['emission'] = ce_emission
+
+    args['area'] = area['the_geom']
+
+    if percentile == 'hist':
+        dates = percentile
+    else:
+        dates = date_range
+    ce_dates = {
+        'hist': ['19710101-20001231'],
+        '2020': ['20100101-20391231',
+                 '20110101-20400101',
+                 '20100101-20391230'],
+        '2050': ['20400101-20691231',
+                 '20410101-20700101',
+                 '20400101-20691230'],
+        '2080': ['20700101-20991231',
+                 '20710101-21000101',
+                 '20700101-20991230']
+    }[dates]
+    args['dates'] = ce_dates
+
+    return args
+
+
+def get_ce_data(sesh, var_name, date_range, area):
+    """Parse a given variable name and into parameters that are used to query
+       the Climate Explorer backend.
+    """
+    try:
+        variable, time_of_year, temporal, \
+            spatial, percentile = var_name.split('_')
+    except ValueError as e:
+        logger.error('Error: Unable to read variable name {} error: {}'
+                     .format(var_name, e))
+        return None
```

Comment:
Again, returning `None` simply propagates the errors down the line to somewhere else where it's going to be less obvious as to what happened.